pkg/kubelet/cm/devicemanager/topology_hints.go

```diff
@@ -66,12 +89,15 @@ func (m *ManagerImpl) deviceHasTopologyAlignment(resource string) bool {
 }
 
 func (m *ManagerImpl) getAvailableDevices(resource string) sets.String {
-	// Gets Devices in use.
-	m.updateAllocatedDevices(m.activePods())
 	// Strip all devices in use from the list of healthy ones.
 	return m.healthyDevices[resource].Difference(m.allocatedDevices[resource])
 }
 
+func (m *ManagerImpl) getAllocatedDevices(podUID, containerName, resource string) sets.String {
+	// Pull the list of device IDs directly from the list of podDevices.
+	return m.podDevices[podUID][containerName][resource].deviceIds
```

Comment:
It's not expected for the 2nd and 3rd level mappings to be nil, but it would be nicer to check (and log and return an empty set I suppose)

Reply:
One better -- there was already a function for this in the `podDevices` abstraction:
```
m.podDevices.containerDevices()
```